beacon_node/lighthouse_network/src/types/sync_state.rs

```diff
@@ -71,6 +74,7 @@ impl SyncState {
             SyncState::BackFillSyncing { .. } => false,
             SyncState::Synced => false,
             SyncState::Stalled => false,
+            SyncState::WaitingOnExecution => false,
```

Comment:
This affects some things like work we do on the beacon processor. Here we are known to be not-synced but not sending batches for processing. I'm not sure of the right value here @paulhauner